src/lib/utils/slug.ts

```diff
@@ -0,0 +1,54 @@
+/**
```

Comment:
i'd almost argue for this to be included in the `utils/listingPages` file instead (i realize it is used in `getStaticProps` instead of `getStaticPaths`, but it feels more like a listingPage utility than a slug utility to me personally)

Reply:
I had this exact same dilemma. I initially wrote it with verbiage specific to listing pages, in fact. But the real catch here is that this function isn't _only_ for listing pages. You can see that it is called for _all_ pages. It is primarily meant to simplify the `getStaticProps` function by removing a ton of conditional logic out of it. I didn't want to bloat that function with a lot of checks for various things. It seemed a _lot_ cleaner to me to simply have a single line of code that would standardize things for all possible slugs (`const [path, pageNumber] = getPathAndPageNumberFromSlug(slug)`).

It might help clarify my point by considering that this function ultimately does this:
```
if (isListingPage(slug)) {
  return getListingPagePathAndPageNumberFromSlug(slug)
} else {
  return [getOtherPagePathFromSlug(slug), undefined]
}
```

It's not written in this fashion exactly, but that's one way to consider what it's doing. If it were written in this way, we could then have the `getListingPagePathAndPageNumberFromSlug` function defined in `utils/listingPages.ts`. 

I considered this approach, but ultimately didn't write the function in that way because I went the route of writing it a bit more general. I didn't want to think specifically of a certain resource type (e.g. story listings). Instead, it made sense to just check for any path that has `page-{n}` in the last slug segment. Since I went this route, I didn't need anything specifically in `utils/listingPages.ts` because, again, the function isn't specific to listing pages. It's called for _every_ page. After all, every page comes in as a slug and needs to be converted to a path.

Reply:
it does run on every path, but it's specifically finding page numbers for the benefit of listing pages. if it isn't a listing page, it returns the path and keeps it moving. in my mind, that logic really feels like a listingPage utility (whether story listing, event listing, etc etc), but i don't think it's a blocker here, just a matter of semantics

Reply:
While I still think the function I wrote would belong outside `utils/listingPages.ts`, this conversation made me reconsider the way I was structuring the logic in the first place. My initial goal of abstracting away the conditional logic around listing pages in `getStaticProps` ultimately felt confusing. The function `getPathAndPageNumberFromSlug` was, indeed, called for _every_ slug, but the page number was ultimately not applicable for non-listing pages. So, I refactored this and introduced some conditional logic based on whether a page is a listing page. I think this ultimately makes the `getStaticProps` function more clear, if slightly more verbose.

Reply:
And, to your initial point @tjheffner, this allows the move of the page-number logic to `utils/listingPages.ts` where it belongs. It's now just in a function that _is_ specific to listing pages.

Reply:
i love the refactor, thank you